### PR TITLE
Amd64/amd64 bringup smp

### PR DESCRIPTION
**Pull request by tkreuzer**
This PR fixes x64 SMP build and boot and makes SMP the default x64 installation.

### CODE REVIEW

File: ntoskrnl/ke/amd64/thrdini.c [hunk 1]
```diff
@@ -184,6 +184,9 @@ KiSwapContextResume(
                      0);
     }
 
+    /* Old thread os no longer busy */
+    OldThread->SwapBusy = 0;
```
Comment (HBelusca) [inline ntoskrnl/ke/amd64/thrdini.c:188]:
```suggestion
    /* Old thread is no longer busy */
    OldThread->SwapBusy = FALSE;
```

File: ntoskrnl/ke/ipi.c [hunk 2]
```diff
@@ -167,7 +167,7 @@ KiIpiServiceRoutine(IN PKTRAP_FRAME TrapFrame,
 
     if (InterlockedBitTestAndReset((PLONG)&Prcb->IpiFrozen, IPI_SYNCH_REQUEST))
     {
-#ifdef _M_ARM
+#if defined(_M_ARM) || defined(_M_AMD64)
```
Comment (binarymaster) [inline ntoskrnl/ke/ipi.c:170]: Nit: `#endif // _M_ARM` also needs update

File: base/setup/lib/settings.c [hunk 3]
```diff
@@ -273,6 +273,10 @@ GetComputerIdentifier(
         return FALSE;
     }
 
+#ifdef _M_AMD64
+    /* On x64 we are l33t and use the MP config by default */
+    ComputerIdentifier = L"X64 MP";
```
Comment (binarymaster) [inline base/setup/lib/settings.c:278]: Hehe :)

File: hal/halx86/apic/halinit.c [hunk 4]
```diff
@@ -41,6 +41,9 @@ HalpInitProcessor(
 VOID
 HalpInitPhase0(IN PLOADER_PARAMETER_BLOCK LoaderBlock)
 {
+    DPRINT1("Using hal: APIC %s %s\n",
```
Comment (binarymaster) [inline hal/halx86/apic/halinit.c:44]:
```suggestion
    DPRINT1("Using HAL: APIC %s %s\n",
```

File: ntoskrnl/CMakeLists.txt [hunk 5]
```diff
@@ -54,7 +54,7 @@ add_pch(ntoskrnl ${REACTOS_SOURCE_DIR}/ntoskrnl/include/ntoskrnl.h "${PCH_SKIP_S
 add_dependencies(ntoskrnl psdk asm)
 add_cd_file(TARGET ntoskrnl DESTINATION reactos/system32 NO_CAB FOR all)
 
-if(BUILD_MP)
+if(BUILD_MP OR (ARCH STREQUAL "amd64"))
```
Comment (Extravert-ir) [inline ntoskrnl/CMakeLists.txt:57]: Just remove the `cmake_dependent_option` from sdk/cmake/config.cmake, so BUILD_MP is turned on by default everywhere

**Comment by Extravert-ir** [comment]
The PR overlaps #3672 in many places. Which one we're going to merge first? :)

**Comment by tkreuzer** [comment]
> 
> 
> The PR overlaps #3672 in many places. Which one we're going to merge first? :)

I don't think it overlaps anywhere. When I noticed that zefklop was working on this, I intentionally removed my tcpip fixes (or hacks) from the PR. The rest shouldn't conflict.

**Comment by zefklop** [comment]
Some of the fixes are already part of #3672

**Pull request merged**